### ISSUE TITLE
SSL Labs test version intolerance does not always work

### ISSUE THREAD

**Issue by ivanr** [ssllabs/ssllabs-scan#385]
[Submitted on behalf of @hannob]

As you may know I'm currently trying to tackle the issue of TLS version intolerance. While the SSL Labs test has a check for version intolerance it doesn't always work.
Pages where this happens are e.g.:
* https://www.ssllabs.com/ssltest/analyze.html?d=paypal.com&s=64.4.250.23
* https://www.ssllabs.com/ssltest/analyze.html?d=paypal.it&s=64.4.250.13
* https://www.ssllabs.com/ssltest/analyze.html?d=pandora.com&s=208.85.40.20&latest

For all of them the test shows "Version Intolerance: No", yet one can't establish a successful connection with an 1.3 handshake. (Please note: sometimes the www and the non-www-version behave differently.)

It seems the handshake here fails later on, your test probably only checks for the first serverhello.

You can test this with chrome canary, which already has preliminary 1.3 support. Install a current version of canary, go to chrome://flags and set "Maximum TLS version enabled" to "TLS 1.3".

**Comment by bhushan5640** [comment]
Handshake fails after client key exchange, change cipher spec stage. Should SSL Labs test for complete handshake? As we only check for the response serverhello - which seems appropriate in this case.

![paypal it](https://cloud.githubusercontent.com/assets/6083185/18159142/fb5b115a-7044-11e6-8e2e-f91ffa5f0dbf.jpg)

**Comment by ivanr** [comment]
Yes, I think so. It seems this is a different problem, perhaps not intolerance, but a bug somewhere in the TLS stack. Still, the result is the same.
